lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
Okay, so here are my thoughts. Instead of adding the separate variable, this logic should be in `guess_galaxy_port()`. https://github.com/jmchilton/galaxy/blob/ie_updates/lib/galaxy/config.py#L645 In the uWSGI specific except case, why not read the galaxy_infrastructure_url and decide 80/443 from http/https there.